sdk/eventhub/Azure.Messaging.EventHubs/samples/Sample09_ObservableEventBatch.md

```diff
@@ -0,0 +1,152 @@
+
+```C# Snippet:Sample09_ObservableEventBatch
+public class ObservableEventDataBatch : IDisposable
+{
+    // The set of events that have been accepted into the batch
+    private List<EventData> _events = new List<EventData>();
+
+    /// The EventDataBatch being observed
+    private EventDataBatch _batch;
+
+    // These events are the source of what is held in the batch.  Though
+    // these instances are mutable, any changes made will NOT be reflected to
+    // those that had been accepted into the batch
+    public IReadOnlyList<EventData> Events { get; }
+
+    public int Count => _batch.Count;
+    public long SizeInBytes => _batch.SizeInBytes;
+    public long MaximumSizeInBytes => _batch.MaximumSizeInBytes;
+
+    // The constructor requires that sourceBatch is an empty batch so that it can track the events
+    // that are being added
+    public ObservableEventDataBatch(EventDataBatch sourceBatch)
+    {
+        _batch = sourceBatch ?? throw new ArgumentNullException(nameof(sourceBatch));
+        if (_batch.Count > 0)
+        {
+            throw new ArgumentException("The sourceBatch is not empty.", nameof(sourceBatch));
+        }
+        Events = _events.AsReadOnly();
+    }
+
+    public bool TryAdd(EventData eventData)
+    {
+        if (_batch.TryAdd(eventData))
+        {
+            _events.Add(eventData);
+            return true;
+        }
+
+        return false;
+    }
+
+    public void Dispose() => _batch.Dispose();
+
+    // Performs the needed transation to allow an ObservableEventDataBatch to be
+    // implicitly converted to an EventDataBatch
+    public static implicit operator EventDataBatch(ObservableEventDataBatch observable) => observable._batch;
+}
+```
+
+#### Benefits
+
+The benefit of this approach is that it utilizes all of the existing methods for sending and publishing events. It takes advantage of the implicit operator functionality in order to return the internal `EventDataBatch` variable, `_batch`, when these methods are called. It also is a straightfoward implementation that achieves the goal of allowing events to be seen by the application.
+
+#### Trade-offs
+
+The trade-off of this approach is that as mentioned above it does not prevent issues where the two disntinct set of events (the actual `EventBatch` events and the visible list of events) go out of sync. There are two ways that this could happen, the first is if the events returned by `ObservableDataBatch.Events` are mutated by the application. The second way this could happen is if `TryAdd` is called on the reference to the `EventDataBatch` after it has been casted from an `ObservableEventDataBatch`, this would call `EventDataBatch.TryAdd()` rather than `ObservableEventDataBatch.TryAdd()` leading to added events being in the actual batch to send that are not reflected in the visible `ObservableDataBatch.Events` variable.
```

Comment:
```suggestion
The trade-off of this approach is that as mentioned above it does not prevent issues where the two distinct set of events (the actual `EventBatch` events and the visible list of events) go out of sync. There are two ways that this could happen, the first is if the events returned by `ObservableDataBatch.Events` are mutated by the application. The second way this could happen is if `TryAdd` is called on the reference to the `EventDataBatch` after it has been casted from an `ObservableEventDataBatch`, this would call `EventDataBatch.TryAdd` rather than `ObservableEventDataBatch.TryAdd` leading to added events being in the actual batch to send that are not reflected in the visible `ObservableDataBatch.Events` variable.
```

Well described here;  this could have been confusing but you managed to make it approachable.